Source/Dafny/Dafny.atg

```diff
@@ -2055,25 +2064,30 @@ FunctionSpec<.List<AttributedExpression/*!*/>/*!*/ reqs, List<FrameExpression/*!
      Contract.Requires(cce.NonNullElements(reads));
      Contract.Requires(decreases == null || cce.NonNullElements(decreases));
      Expression/*!*/ e;  FrameExpression/*!*/ fe;
-     Attributes ensAttrs = null; Attributes reqAttrs = null; .)
+     Attributes ensAttrs = null; Attributes reqAttrs = null;
+  .)
   SYNC
   ( "requires"
     { Attribute<ref reqAttrs> }
-    Expression<out e, false, false> OldSemi                (. reqs.Add(new AttributedExpression(e, reqAttrs)); .)
-  | "reads"
+    Expression<out e, false, false> OldSemi             (. reqs.Add(new AttributedExpression(e, reqAttrs)); .)
+  | "reads"                                             (. Attributes readsAttrs = null; .)
+    { Attribute<ref readsAttrs> }
     PossiblyWildFrameExpression<out fe, false>          (. reads.Add(fe); .)
     { "," PossiblyWildFrameExpression<out fe, false>    (. reads.Add(fe); .)
     }
     OldSemi
   | "ensures"
-      { Attribute<ref ensAttrs> }
-      Expression<out e, false, false> OldSemi                (. ens.Add(new AttributedExpression(e, ensAttrs)); .)
+    { Attribute<ref ensAttrs> }
+    Expression<out e, false, false> OldSemi             (. ens.Add(new AttributedExpression(e, ensAttrs)); .)
   | "decreases"                               (. if (decreases == null) {
                                                    SemErr(t, "'decreases' clauses are meaningless for greatest predicates, so they are not allowed");
```

Comment:
I'm guessing this error message should say "greatest and least predicates", not just "greatest predicates".

Reply:
Done